src/higlass-schema/higlass.schema.ts

```diff
@@ -100,21 +100,38 @@ export interface EnumTrack {
     y?: number;
     options?: any;
 }
+
 export interface Data {
-    type?: string; // TODO: What kinds of types exist?
+    type?: string;
     children?: any[];
     tiles?: any;
     tilesetInfo?: any;
     url?: string;
 
     // Options Gosling internally use
-    assembly?: Assembly;
 
+    assembly?: Assembly;
+    urlFetchOptions?: RequestInit; // Options used when URL is fetched
+    indexUrl?: string; // For datatypes that have indexes
+    indexUrlFetchOptions?: RequestInit; // Options used when index URL is fetched
     // Option to filter datasets
     // This has been added to filter data properly during fetching tiles
     filter?: FilterTransform[];
 }
 
+// When using the built-in RequestInit type, `yarn schema` doesn't work. So we are using our own RequestInit.
+export interface RequestInit {
+    headers?: Record<string, any>;
+    body?: any;
+    method?: string;
+    mode?: string;
+    credentials?: string;
+    cache?: string;
+    referrer?: string;
+    referrerPolicy?: string;
+    [key: string]: any;
+}
+
```

Comment:
Should I add the [whole](https://fetch.spec.whatwg.org/#requestinit) RequestInit type here? Or just a subset?

Reply:
I think the subset that we actually use should be fine